modules/nextflow/src/main/groovy/nextflow/data/cid/CidHistoryRecord.groovy

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2013-2024, Seqera Labs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package nextflow.data.cid
+
+import groovy.transform.EqualsAndHashCode
+
+import java.text.DateFormat
+import java.text.SimpleDateFormat
+
+/**
+ * Record of workflow executions and their corresponding CIDs
+ *
+ * @author Jorge Ejarque <jorge.ejarque@seqera.io>
+ */
+@EqualsAndHashCode(includes = 'runName,sessionId')
+class CidHistoryRecord {
+    public static final DateFormat TIMESTAMP_FMT = new SimpleDateFormat('yyyy-MM-dd HH:mm:ss')
+    Date timestamp
+    String runName
+    UUID sessionId
+    String runCid
```

Comment:
Consider making those final and add constructor to initialise it (instead of implicit map-based one)

Reply:
Could all non mutable fields be declared `final`?

Reply:
Changed